src/diffusers/optimization.py

```diff
@@ -34,6 +34,7 @@ class SchedulerType(Enum):
     POLYNOMIAL = "polynomial"
     CONSTANT = "constant"
     CONSTANT_WITH_WARMUP = "constant_with_warmup"
+    CONSTANT_WITH_RULES = "constant_with_rules"
```

Comment:
I might not be familiar enough with this scheduler, but "constant with rules" does not mean a lot to me. Could we use something like "piecewise constant" or "constant with steps"?

Reply:
Piecewise constants sounds better and in line with what I have seen being used in the other libraries too.

Reply:
You are right. I should change the name to piecewise constant.